internal/provider/resource_consul_cluster.go

```diff
@@ -382,12 +387,20 @@ func resourceConsulClusterCreate(ctx context.Context, d *schema.ResourceData, me
 		return diag.Errorf("unable to retrieve Consul cluster (%s): %v", payload.Cluster.ID, err)
 	}
 
+	if err := setConsulClusterResourceData(d, cluster); err != nil {
```

Comment:
When the create fails, where would the tf have failed previously? Was it in attempting to get the client config files?

I had assumed the WaitForOperation call back on line 379 still prior to this call.

Reply:
you're right and yes, this was failing up above in the Wait. Where I saw this was an issue was in `resourceConsulClusterRead` (the `GetConsulClusterByID` call succeeds while `GetConsulClientConfigFiles` fails). I then applied the same pattern everywhere we make these two API calls (storing results of first)

Reply:
Yes, we'll still clean it up on the next `terraform delete` or `terraform apply`. A PR from last year addressed this by storing the ID of the resource before polling: https://github.com/hashicorp/terraform-provider-hcp/pull/59

Current behavior:
- `tf create` fail during poll: bail, return diag.Error, we store nothing but ID
- `tf plan`, read: GET with ID, see it's FAILED, purge from state
- `tf apply`: create a _new_ cluster. Don't delete prior cluster. Run into a duplicate cluster/hvn ID issue

Behavior after this change:
- `tf create` fail during poll: bail, return diag.Error, we store nothing but ID
- `tf plan`, read: GET with ID, set state
- `tf apply`, old cluster is tainted, delete, create a _new_ cluster

Reply:
I realized and messed up and replied out of thread: https://github.com/hashicorp/terraform-provider-hcp/pull/326#issuecomment-1155493682